src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -655,29 +627,22 @@ private void ReadGnuLongPathDataBlock(Stream archiveStream)
             }
         }
 
-        // Asynchronously reads the long path found in the data section of a GNU entry of type 'K' or 'L'
-        // and replaces Name or LinkName, respectively, with the found string.
-        // Throws if end of stream is reached.
-        private static async ValueTask<string?> ReadGnuLongPathDataBlockAsync(Stream archiveStream, TarEntryType entryType, long size, CancellationToken cancellationToken)
+        // Return a byte array if the size field has a valid value for GNU long metadata entry data. Otherwise, return null, or throw.
+        private byte[]? CreateGnuLongDataBufferIfSizeIsValid()
```

Comment:
a very similar logic is performed as part of `CreateExtendedAttributesBufferIfSizeIsValid` method, we could reduce the code duplication by having a single helper for that

Reply:
Since it's also related to extended attributes handling suggestion you've been giving, I can also address this in another PR. I might have to refactor that code a lot and maybe the buffer creation method won't be necessary in the end.